source/speech/speech.py

```diff
@@ -1374,8 +1383,14 @@ def speakTypedCharacters(ch: str):
 		clearTypedWordBuffer()
 		if log.isEnabledFor(log.IO):
 			log.io("typed word: %s" % typedWord)
-		if config.conf["keyboard"]["speakTypedWords"] and not typingIsProtected:
-			speakText(typedWord)
+		typingEchoMode = config.conf["keyboard"]["speakTypedWords"]
+		if typingEchoMode > TypingEcho.OFF.value and not typingIsProtected:
+			if (
+				typingEchoMode == TypingEcho.EDIT_CONTROLS.value
+				and is_editableControl()
+				or typingEchoMode == TypingEcho.EVERYWHERE.value
+			):
```

Comment:
In general, you should perform the most expensive checks first when using `or`, as Python knows that if the first condition is true, it doesn't need to check the second condition. While this check shouldn't be too costly, it is run so often that this may make a measurable difference over time.
Also added an extra set of brackets for readability.
```suggestion
			if (
				typingEchoMode == TypingEcho.EVERYWHERE.value
				or (
					typingEchoMode == TypingEcho.EDIT_CONTROLS.value
					and is_editableControl()
				)
			):
```